pkg/envoy/ads/types.go

```diff
@@ -18,10 +19,18 @@ var (
 	log = logger.New("envoy/ads")
 )
 
+// XdsResponseFunction is a function type entry point for xDS verticals.
+// Params:
+// - It gets OSM configuration state provided by catalog, configurator and certificate manager interfaces
+// - It gets references to the Proxy and the associated discovery request that triggered the execution of the given function (segregated by TypeURI)
+// Returns:
+// - Returns a list of resources composed by the related request and TypeURL call.
+type XdsResponseFunction func(catalog.MeshCataloger, *envoy.Proxy, *xds_discovery.DiscoveryRequest, configurator.Configurator, certificate.Manager) ([]types.Resource, error)
```

Comment:
It seems to me that this is only used within `pkg/envoy/ads` so we don't need to export it (so lower case xds could be better).

Does it make sense to also pull this type in a separate PR since it seems orthogonal to some of the other changes in this PR?

Reply:
yes, absolutely. I'll push it separately.